src/bugsnag.js

```diff
@@ -427,22 +453,83 @@
     // functional fu to make it easier to setup event listeners
     function wrapBuilder(builder) {
       return function() {
+        if(!getBreadcrumbSetting("autoBreadcrumbsNavigation", true)) {
+          return;
+        }
+
         self.leaveBreadcrumb(builder.apply(null, arguments));
       };
     }
 
+    // check for browser support
+    if (!_hasAddEventListener ||
+        !window.history ||
+        !window.history.state ||
+        !window.history.pushState ||
+        !window.history.pushState.bind
+    ) {
+      return;
+    }
+
+    // create enable function
+    self.enableAutoBreadcrumbsNavigation = function() {
+      self.autoBreadcrumbsNavigation = true;
+      // keep track of native functions
+      nativePushState = history.pushState;
+      nativeReplaceState = history.replaceState;
+      // create hooks for pushstate and replaceState
+      enhance(history, "pushState", wrapBuilder(buildPushState));
+      enhance(history, "replaceState", wrapBuilder(buildReplaceState));
+    };
+
+    // create disable function
+    self.disableAutoBreadcrumbsNavigation = function() {
+      self.autoBreadcrumbsNavigation = false;
+      // restore native functions
+      history.pushState = nativePushState;
+      history.replaceState = nativeReplaceState;
+    };
+
     window.addEventListener("hashchange", wrapBuilder(buildHashChange), true);
     window.addEventListener("popstate", wrapBuilder(buildPopState), true);
     window.addEventListener("pagehide", wrapBuilder(buildPageHide), true);
     window.addEventListener("pageshow", wrapBuilder(buildPageShow), true);
     window.addEventListener("load", wrapBuilder(buildLoad), true);
     window.addEventListener("DOMContentLoaded", wrapBuilder(buildDOMContentLoaded), true);
 
-    // create hooks for pushstate and replaceState
-    enhance(history, "pushState", wrapBuilder(buildPushState));
-    enhance(history, "replaceState", wrapBuilder(buildReplaceState));
+    if(getBreadcrumbSetting("autoBreadcrumbsNavigation", true)) {
+      self.enableAutoBreadcrumbsNavigation();
+    }
   }
 
+  self.enableAutoBreadcrumbsErrors = function() {
+    self.autoBreadcrumbsErrors = true;
+  };
+
+  self.disableAutoBreadcrumbsErrors = function() {
+    self.autoBreadcrumbsErrors = false;
+  };
+
+  self.enableAutoBreadcrumbsClicks = function() {
+    self.autoBreadcrumbsClicks = true;
+  };
+
+  self.disableAutoBreadcrumbsClicks = function() {
+    self.autoBreadcrumbsClicks = false;
+  };
+
+  self.enableAutoBreadcrumbs = function() {
+    self.enableAutoBreadcrumbsClicks();
+    self.enableAutoBreadcrumbsConsole();
+    self.enableAutoBreadcrumbsNavigation();
```

Comment:
Just curious — how come we don't have to call `enableAutoBreadcrumbsErrors` here?